author: Andres Salomon <dilinger@debian.org>

Freetype 2.12 wasn't supposed to have COLRV1 enabled, but it was enabled
in bookworm. Then the maintainer issued a stable update to disable it,
which caused chromium to crash because we're actually using it. But
apparently we're not supposed to be, so bump this version check from 2.11
to 2.13 (and mangle that FT_STATIC_CAST check).

See https://bugs.debian.org/1052455 for the description about freetype 2.12
having a broken COLRV1.

--- a/third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
+++ b/third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
@@ -56,9 +56,9 @@ namespace {
 // FT_STATIC_CAST introduced VER-2-11-0-172-g9079c5d91
 // So undefine TT_SUPPORT_COLRV1 before 2.11.1 but not if FT_STATIC_CAST is defined.
 #if (((FREETYPE_MAJOR)  < 2) || \
-     ((FREETYPE_MAJOR) == 2 && (FREETYPE_MINOR)  < 11) || \
+     ((FREETYPE_MAJOR) == 2 && (FREETYPE_MINOR)  < 13) || \
      ((FREETYPE_MAJOR) == 2 && (FREETYPE_MINOR) == 11 && (FREETYPE_PATCH) < 1)) && \
-    !defined(FT_STATIC_CAST)
+    !defined(FT_ATIC_CAST)
 #    undef TT_SUPPORT_COLRV1
 #else
 #    include "src/base/SkScopeExit.h"
--- a/third_party/skia/src/ports/SkFontHost_FreeType.cpp
+++ b/third_party/skia/src/ports/SkFontHost_FreeType.cpp
@@ -91,9 +91,9 @@ using SkUniqueFTSize = std::unique_ptr<F
 // FT_STATIC_CAST introduced VER-2-11-0-172-g9079c5d91
 // So undefine TT_SUPPORT_COLRV1 before 2.11.1 but not if FT_STATIC_CAST is defined.
 #if (((FREETYPE_MAJOR)  < 2) || \
-     ((FREETYPE_MAJOR) == 2 && (FREETYPE_MINOR)  < 11) || \
+     ((FREETYPE_MAJOR) == 2 && (FREETYPE_MINOR)  < 13) || \
      ((FREETYPE_MAJOR) == 2 && (FREETYPE_MINOR) == 11 && (FREETYPE_PATCH) < 1)) && \
-    !defined(FT_STATIC_CAST)
+    !defined(FT_ATIC_CAST)
 #    undef TT_SUPPORT_COLRV1
 #endif
 #endif
